### PR TITLE
Add option to generate omnibor artifact IDs.

### PR DESCRIPTION
**Pull request by duckinator**

### CODE REVIEW

**Comment by duckinator** [comment]
Current state:

- test suite runs locally
- real-world test ([duckinator/dist-sometest v0.0.14](https://github.com/duckinator/dist-sometest/releases/tag/v0.0.14)) works
- CI fails, I believe because omnibor-cli needs to be installed in the ci.yml workflow

**Comment by duckinator** [comment]
Alright, this should be ready to go!

- test suite passes locally
- real-world test ([duckinator/dist-sometest v0.0.14](https://github.com/duckinator/dist-sometest/releases/tag/v0.0.14)) works
- CI passes

**Comment by ashleygwilliams** [comment]
@alilleybrinker would be curious if you have any thoughts on this!

**Comment by alilleybrinker** [comment]
> @alilleybrinker would be curious if you have any thoughts on this!

Taking a look!

**Review by alilleybrinker — APPROVED** [review]
LGTM! I see you settled on a `.omnibor` file containing the Artifact ID, which seems fine to me. I don't think the OmniBOR project has a consensus on the naming of this file, and this seems as good a solution as any!

File: cargo-dist/src/lib.rs
Comment (mistydemeo) [inline cargo-dist/src/lib.rs:428]: How would you feel about adding `omnibor` to the `Tools` struct? That would make it easier for us to handle the "omnibor not installed" case: https://github.com/axodotdev/cargo-dist/blob/main/cargo-dist/src/tasks.rs#L264-L282

File: cargo-dist/src/tasks.rs
Comment (mistydemeo) [inline cargo-dist/src/tasks.rs:1561]: Nit: if this is written as an `if matches!(artifact.kind, ...)` condition, you could skip the unwanted `ArtifactKind` cases and only handle the desired ones.

File: cargo-dist/src/config/v0.rs
Comment (mistydemeo) [inline cargo-dist/src/config/v0.rs:474]: I wonder if we want to name this something more specific? Like `generate-omnibor-id`.

**Review by mistydemeo — COMMENTED** [review]
Looks good, just a couple comments.

File: cargo-dist/src/tasks.rs
Comment (duckinator) [reply cargo-dist/src/tasks.rs:1561]: This was actually deliberate, because it means if another `ArtifactKind` is added the compiler will tell us we need to decide where it goes, instead of just assuming all new `ArtifactKind`s aren't relevant.

If you think I should change it anyway, lmk and I will. Otherwise I'll add a comment explaining why I did it.

Comment (mistydemeo) [reply cargo-dist/src/tasks.rs:1570]: Ah, good call! That makes sense.

**Review by mistydemeo — APPROVED** [review]
Looking good, though we should decide on the final config option before this is merged. Also looks like the tests need to be adjusted for the new field in the `Tools` struct.

**Pull request merged**
